### PR TITLE
test: add direct unit tests for app/services/honeycomb/client.py 

### LINKED ISSUE #883: Add direct unit tests for `app/services/honeycomb/client.py`
## Why This Matters
Honeycomb query creation and polling are core client behavior, but current coverage is mostly indirect.

## Read First
- `app/services/honeycomb/client.py`
- `tests/cli/wizard/test_integration_health.py`

## Scope
1. Add a service test module for Honeycomb.
2. Cover `validate_access()`, query creation, and query-result retrieval.
3. Cover both HTTP status failures and generic exceptions.

## Tests / Coverage
- `tests/services/test_honeycomb_client.py`
- cover auth validation
- cover `create_query()`
- cover `create_query_result()`
- cover `get_query_result()`

## Acceptance Criteria
- Honeycomb client behavior is directly tested
- failures are handled consistently
- no network access is required

### PR DESCRIPTION
**Pull request by gitsofaryan**
Fixes #883

#### Describe the changes you have made in this PR -
Added direct unit tests for the Honeycomb service client in `app/services/honeycomb/client.py`. These tests cover key behaviors including authentication validation, query creation, and query result retrieval (polling). The implementation uses `unittest.mock` to ensure no network access is required, and covers both success paths and error handling for HTTP failures and generic exceptions.

### Demo/Screenshot for feature changes and bug fixes - 
```text
tests/services/test_honeycomb_client.py::test_is_configured PASSED      [  7%]
tests/services/test_honeycomb_client.py::test_validate_access_success PASSED [ 14%]
tests/services/test_honeycomb_client.py::test_validate_access_http_error PASSED [ 21%]
tests/services/test_honeycomb_client.py::test_validate_access_generic_exception PASSED [ 28%]
tests/services/test_honeycomb_client.py::test_create_query_success PASSED [ 35%]
tests/services/test_honeycomb_client.py::test_create_query_no_id PASSED   [ 42%]
tests/services/test_honeycomb_client.py::test_create_query_http_error PASSED [ 50%]
tests/services/test_honeycomb_client.py::test_create_query_result_success PASSED [ 57%]
tests/services/test_honeycomb_client.py::test_create_query_result_http_error PASSED [ 64%]
tests/services/test_honeycomb_client.py::test_get_query_result_success PASSED [ 71%]
tests/services/test_honeycomb_client.py::test_get_query_result_http_error PASSED [ 78%]
tests/services/test_honeycomb_client.py::test_get_query_result_generic_exception PASSED [ 85%]
tests/services/test_honeycomb_client.py::test_run_query_full_flow PASSED  [ 92%]
tests/services/test_honeycomb_client.py::test_query_traces_success PASSED [100%]

============================= 14 passed in 3.82s ==============================
```

---

## Code Understanding and AI Usage

**Did you use AI assistance (ChatGPT, Claude, Copilot, etc.) to write any part of this code?**
- [x] Yes, I used AI assistance (Antigravity)

**If you used AI assistance:**
- [x] I have reviewed every single line of the AI-generated code
- [x] I can explain the purpose and logic of each function/component I added
- [x] I have tested edge cases and understand how the code handles them
- [x] I have modified the AI output to follow this project's coding standards and conventions

**Explain your implementation approach:**
The tests use `pytest` and `unittest.mock` to isolate the `HoneycombClient` from the external API and the `httpx` library. I focused on covering all branch points in the exception handling logic to ensure that HTTP errors (4xx/5xx) and network-level failures are handled consistently and return the expected error structure. High-level methods like `run_query` and `query_traces` are also tested to ensure correct internal logic and payload formatting.

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue
- [x] I have performed a self-review of my code
- [x] **I can explain the purpose of every function, class, and logic block I added**
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [x] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

### CODE REVIEW

**Comment by gitsofaryan** [comment]
cc : @VaibhavUpreti @rrajan94 PTAL!

**Comment by gitsofaryan** [comment]
@muddlebee PTAL!

**Comment by muddlebee** [comment]
@gitsofaryan pls fix the greptile review points?

**Comment by gitsofaryan** [comment]
@muddlebee  addressed!

File: tests/services/test_honeycomb_client.py [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+
+    assert result["success"] is True
+    assert result["query_id"] == "q1"
+    assert result["query_result_id"] == "r1"
+    assert result["results"] == [{"val": 10}]
+
+
+def test_query_traces_success(client: HoneycombClient) -> None:
+    client.run_query = MagicMock(return_value={"success": True, "results": []})
+
```
[the remark this answers is not in the document]
Comment (muddlebee) [reply tests/services/test_honeycomb_client.py:224]: @gitsofaryan not addressed yet

**Comment by muddlebee** [comment]
@gitsofaryan CI still failing

**Comment by gitsofaryan** [comment]
@muddlebee fixed!

**Comment by muddlebee** [comment]
@gitsofaryan great work. thank you :guitar:
